operator/pkg/model/translation/gateway-api/translator.go

```diff
@@ -180,6 +188,39 @@ func getEndpoints(resource model.FullyQualifiedResource) *corev1.Endpoints {
 	}
 }
 
+func decorateCEC(cec *ciliumv2.CiliumEnvoyConfig, resource *model.FullyQualifiedResource, labels, annotations map[string]string) (*ciliumv2.CiliumEnvoyConfig, error) {
+	if cec == nil || resource == nil {
+		return cec, fmt.Errorf("CEC or resource can't be nil")
+	}
+	res := cec.DeepCopy()
+
+	// Set the owner reference to the CEC object.
+	res.OwnerReferences = []metav1.OwnerReference{
+		{
+			APIVersion: resource.Group + "/" + resource.Version,
+			Kind:       resource.Kind,
+			Name:       resource.Name,
+			UID:        types.UID(resource.UID),
+			Controller: model.AddressOf(true),
+		},
+	}
+
+	shortedName := model.Shorten(resource.Name)
+	if len(res.Labels) == 0 {
```

Comment:
don't you actually want to test for the `nil` value with `if res.Labels == nil`?